server/routes/users/index.ts

```diff
@@ -22,4 +49,13 @@ app.delete('/api/v1/logout', validateRefreshJWT, wrap(userController.logout));
 // get user reactions
 app.get('/api/v1/users/reactions', validateRefreshJWT, wrap(userController.getUserReactions));
 
+// get user data such as reputation and number of posts
+app.get('/api/v1/users', validateRefreshJWT, wrap(userController.getUserData));
+
+// change the user image
+app.post('/api/v1/users/change-icon', validateRefreshJWT, upload.single('image'), wrap(userController.changeUserIcon));
+
+// get the icon of a user
+app.get('/api/v1/users/icon', wrap(userController.getUserIcon));
```

Comment:
Here you have several APIs and all of them are around a single user. And it is not clear from the route.
When you work with a single object you may specify the specific object in a route as `users/:id/...`.
Because how would you create an endpoint to get the list of all users right now?

And one more topic related is where do you use cookies. You may use it for validation - if this user has the right to perform a particular action. But you can't make an assumption that the user wants to perform an action under his own instance. 
It could be a situation where you may have the catalog of all users of the public user page, where I can go to the link and get your icon, post number, and reputation.